nb-api/src/main/java/io/nosqlbench/nb/api/content/StreamContent.java

```diff
@@ -0,0 +1,53 @@
+package io.nosqlbench.nb.api.content;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.CharBuffer;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * StreamContent is meant for short-lived use as an efficient way to
+ * find a read URL content. If a caller has already acquired an
+ * input stream, it can be passed to the stream content holder
+ * to avoid double fetch or other unintuitive and inefficient
+ * behavior.
+ *
+ * Subclasses must implement getURI()
+ */
+public abstract class StreamContent<T> implements Content<T> {
+    private final T location;
+    private final InputStream inputStream;
+
+    public StreamContent(T location, InputStream inputStream) {
+        this.location = location;
+        this.inputStream = inputStream;
+    }
+
+    @Override
+    public T getLocation() {
+        return location;
+    }
+
+    @Override
+    public CharBuffer getCharBuffer() {
+        InputStreamReader inputStreamReader = new InputStreamReader(inputStream, StandardCharsets.UTF_8);
```

Comment:
This is missing the lazy-init and cache behavior of the previous implementation. This means that that readers will exhaust the stream when read a second time. Please cache and return the buffer instead. These objects will be disposed once out of scope anyway.